### PR TITLE
Navigation List view: Include offcanvas specific styles

### PR DESCRIPTION
**Pull request by MaggieCabrera**
## What?

This PR cherry picks https://github.com/WordPress/gutenberg/pull/45947/commits/a273d65669c73b5276d40304086f26d0ab7876e4 from #45947 so the new off canvas editor includes its own CSS file

## Why?

Originally we were duplicating the list view code and not including the copied scss file. We are cleaning up here and including it.

### CODE REVIEW

File: packages/block-editor/src/components/off-canvas-editor/style.scss [hunk 1]
```diff
@@ -1,397 +1,17 @@
...
-	pointer-events: none;
-
-	.block-editor-list-view-drop-indicator__line {
-		background: var(--wp-admin-theme-color);
-		height: $border-width;
+	&:hover,
+	&:focus {
+		box-shadow: none;
 	}
 }
```
Comment (getdave) [inline packages/block-editor/src/components/off-canvas-editor/style.scss]: Shall we just include an empty file to start with as these changes don't make sense in the context of this PR 🙏

**Review by scruffian — APPROVED** [review]
LGTM

**Review by draganescu — APPROVED** [review]

**Review by getdave — APPROVED** [review]
Thank you for pulling this PR out for us 🙏 

It's required because on `trunk` the styles for the offcanvas list view weren't being included even though the `style.scss` file was in the file tree.

We should be extremely careful and selective about the styles we introduce into this custom file as anything we do may need to be merge back into the core list view later.

File: packages/block-editor/src/components/off-canvas-editor/style.scss (hunk 1 above)
Comment (MaggieCabrera) [reply packages/block-editor/src/components/off-canvas-editor/style.scss]: makes sense to me, I'll do that

**Review by getdave — CHANGES_REQUESTED** [review]
I'm blocking this merge temporarily because I'd like the changes in https://github.com/WordPress/gutenberg/pull/45963#discussion_r1029144227 to be actioned prior to merge.

File: packages/block-editor/src/components/off-canvas-editor/style.scss [hunk 2]
```diff
@@ -1,397 +1 @@
...
-	}
-}
-
-.block-editor-list-view-placeholder {
-	padding: 0;
-	margin: 0;
-	height: 36px;
-}
-
+//Styles for off-canvas editor, remove this line when you add some css to this file!
```
Comment (getdave) [inline packages/block-editor/src/components/off-canvas-editor/style.scss:1]: 💥

**Review by getdave — APPROVED** [review]
:shipit:

**Pull request merged**